cmd/shim/main.go

```diff
@@ -408,9 +434,10 @@ func setupBundle() int {
 			// provide the server id to connect back to
 			if execMetadata.ServerID == "" {
 				fmt.Fprintln(os.Stderr, "missing server id")
-				return 1
+				return errorExitCode
 			}
 			keepEnv = append(keepEnv, "_DAGGER_SERVER_ID="+execMetadata.ServerID)
+			keepEnv = append(keepEnv, "_DAGGER_ENVIRONMENT_DIGEST="+execMetadata.EnvironmentDigest.String())
```

Comment:
TBH I hate the complication this added to the code (`CurrentEnvironment` is inherently complicated since it is self-referential and thus creates circular dependencies unless we are very careful), but yeah, it's neat now that it works! I'm also hopeful that we can find some simpler patterns for this in a future iteration, it felt like something that could be improved with another round of thought/eyes.